docs/class_diagram.pu

```diff
@@ -0,0 +1,228 @@
+}
+
+class PlaceData{
+__ private data __
+- id : number
+- address : string
+- city : string
+- zipcode : string
+- state : string
+==
++ PlaceData(address : string, city : string, zipcode : string, state : string)
++ PlaceData(id : number, address : string, city : string, zipcode : string, state : string)
++ getId()
++ getAddress()
++ getCity()
++ getZipcode()
++ getState()
++ setAddress(address : string)
++ setCity(city : string)
++ setZipcode(zipcode : string)
++ setState(state : string)
+}
+
+
+class Organization{
+  __ private data __
+  - ID : integer
+  - organizationName : string
+  - known : bool
+  - places : Place[]
+  - description : string
+  - usersRole : UsersRole[]
+  ==
+  + Organization(name : string, known : bool, description : string)
+  + Organization(number: id, name : string, known : bool, places : Place[], description : string, usersRole : UsersRole[])
+  + getID()
+  + getOrganizationName()
+  + getKnown()
+  + getPlaces()
+  + getDescription()
+  + setOrganizationName(organizationName : string)
+  + setKnown(known : bool)
+  + setDescription(description : string)
+  + setPlaces(places : Place[])
+}
+
+class LoginComponent {
+  __ private data __
+  - User : user
+  ==
+  + constructor(auth : AuthenticationService)
+  + login()
+  + loginError()
+}
+
+class environment{
+  + apiUrl : string
+}
+class AuthenticationService {
+  __ private data __
+  - clientId : number
+  ==
+  + constructor(http : HttpClient)
+  + login(user : User)
+  + checkCookie()
+  + logout()
+  + getToken(code : any)
+  + saveToken()
+}
+
+class HomePageComponent {
+ + constructor(auth : AuthenticationService)
+}
+
+class TopBarComponent {
+ __ private data __
+ - User : user
+ ==
+ + constructor(auth : AuthenticationService, userService : UserService)
+ + getUser()
+ + getDate()
+ + getTime()
+}
+
+class ProfileComponent {
+  __ private data __
+  - User : user
+  ==
+  + constructor(auth : AuthenticationService, userService : UserService)
+  + getUser()
+}
+
+class UserService {
+  + constructor()
+  + getUser()
+}
+
+class ShowOrganizationComponent {
+  __ private data __
+  - organizations : Organization[]
+  ==
+  + constructor(auth : AuthenticationService, userService : UserService, organizationService : OrganizationService)
+  + getOrganizations( userId : number)
+}
+
+class OrganizationRequestComponent {
+  + constructor(auth : AuthenticationService, userService : UserService, organizationService : OrganizationService)
+  + sendRequest( organization : Organization)
+}
+
+class OrganizationService {
+ + constructor()
+ + getOrganization( userId : number)
+ + createOrganization( organization : Organization)
+}
+
+class MonitorOrganizationComponent
+class ShowEmployeesComponent
+class ShowPlacesComponent
+class ManagePlacesComponent
+class ManageOrganizationsComponent
+class ManageUsersComponent
+
+class OrganizationListComponent
+class OrganizationListService
+class OrganizationDetailsComponent
+class OrganizationDetailsService
```

Comment:
queste sono classi incomplete? comunque questo diagramma è troppo consistente, va spezzato in più parti